python/cog/director/director.py

```diff
@@ -156,17 +147,19 @@ def _setup(self):
 
             return
 
-    def _loop(self):
+    def _loop(self) -> None:
         while not self._should_exit:
+            redis_consumer = self.redis_consumer_rotator.get_current(and_rotate=True)
```

Comment:
I imagined `RedisConsumerRotator` providing the same interface as `RedisConsumer`, so the director doesn't have to care which it's given. A call to `get()` could rotate and then query redis. Calls to `ack()` and `checker()` could operator against the current queue. The only thing that breaks is the bound contextvar for redis input queue, but that could move down to after we've called `get()` with minimal impact.

The advantage of that matched interface, IMO, is that understanding the main loop of director doesn't get any more complicated.

Reply:
I started with that approach, but I worry about `get()` having a side effect. Do we definitely always want to rotate after the `get()` call?

Reply:
Oh. Damn, you're right. I'll make it explicit.